lua/lazyvim/plugins/extras/lang/java.lua

```diff
@@ -24,102 +24,134 @@ return {
-              require("which-key").register({ c = { x = { name = "Extract" } } }, { prefix = "<leader>" })
-            end,
-          })
           return true -- avoid duplicate servers
         end,
       },
     },
   },
+
+  -- Set up nvim-jdtls
+  {
+    "mfussenegger/nvim-jdtls",
+    dependencies = { "folke/which-key.nvim" },
+    ft = { "java" },
+    config = function()
+      -- The configuration for jdtls contains two useful items:
+      -- 1. The list of filetypes on which to match.
+      -- 2. Custom method for finding the root for a java project.
+      local lsp_config = require("lspconfig.server_configurations.jdtls").default_config
+      local find_java_project_root = lsp_config.root_dir
+      local filetypes = lsp_config.filetypes
+      -- lookup paths for java test and debugger package
+      local mason_registry = require("mason-registry")
+      local bundles = {}
+      if mason_registry.has_package("java-test") and mason_registry.has_package("java-debug-adapter") then
+        -- jdtls tools configuration for debugging support
+        local java_test_pkg = mason_registry.get_package("java-test")
+        local java_test_path = java_test_pkg:get_install_path()
+        local java_dbg_pkg = mason_registry.get_package("java-debug-adapter")
+        local java_dbg_path = java_dbg_pkg:get_install_path()
+        local jar_patterns = {
+          java_dbg_path .. "/extension/server/com.microsoft.java.debug.plugin-*.jar",
+          java_test_path .. "/extension/server/*.jar",
+        }
+        for _, jar_pattern in ipairs(jar_patterns) do
+          for _, bundle in ipairs(vim.split(vim.fn.glob(jar_pattern), "\n")) do
+            table.insert(bundles, bundle)
+          end
+        end
+      end
+
+      local fname = vim.api.nvim_buf_get_name(0)
+      local root_dir = find_java_project_root(fname)
+      local project_name = root_dir and vim.fs.basename(root_dir)
+      local cmd = { "jdtls" }
+      if project_name then
+        local jdtls_cache_dir = vim.fn.stdpath("cache") .. "/jdtls/" .. project_name
+        vim.list_extend(cmd, {
+          "-configuration",
+          jdtls_cache_dir .. "/config",
+          "-data",
+          jdtls_cache_dir .. "/workspace",
+        })
+      end
+      local jdtls_base_config = {
+        cmd = cmd,
+        root_dir = root_dir,
+        init_options = {
+          bundles = bundles,
+        },
+        -- enable CMP capabilities
+        capabilities = require("cmp_nvim_lsp").default_capabilities(),
+      }
+
+      -- Attach jdtls for the proper filetypes (i.e. java).
+      -- Existing server will be reused if the root_dir matches.
+      vim.api.nvim_create_autocmd("FileType", {
+        pattern = filetypes,
+        callback = function()
+          local jdtls_opts = require("lazyvim.util").opts("nvim-jdtls")
+          require("jdtls").start_or_attach(vim.tbl_deep_extend("force", jdtls_opts or {}, jdtls_base_config))
```

Comment:
This is from the original code, not something you changed. Nonetheless, it should be

```lua
require("jdtls").start_or_attach(vim.tbl_deep_extend("force", jdtls_base_config, jdtls_opts or {})
```

so that local opts can override the base config, rather than the other way around.